### PR TITLE
Qualify function when printing MethodInstance

### PR DESCRIPTION
**Pull request by timholy**
Also print more info for the case `l.def::Module`.

The module qualification builds off #38596. I've wanted the module-qualifier for a long time, but I recognize it might be bad to add it so close to a release. If desired we can wait until 1.7.

### CODE REVIEW

File: base/show.jl [hunk 1]
```diff
@@ -984,18 +984,29 @@ function sourceinfo_slotnames(src::CodeInfo)
...
             show(io, def)
         else
             print(io, "MethodInstance for ")
-            show_tuple_as_call(io, def.name, l.specTypes)
+            show_tuple_as_call(io, def.name, l.specTypes, false, nothing, nothing, true)
         end
     else
         print(io, "Toplevel MethodInstance thunk")
+        # `thunk` is not very much information to go on. If this
+        # MethodInstance is part of a stacktrace, it get location info
```
Comment (Sacha0) [inline base/show.jl:1002]: `it gets` I think? :)

**Review by Sacha0 — APPROVED** [review]
Looks great --- thanks Tim! :)

**Comment by timholy** [comment]
So, is it better to merge or hold off? I'm fine with either; it's more a question of whether it's better to make as many breaking changes to printing in one release as we can (https://julialang.slack.com/archives/C67TK21LJ/p1607230631257400) or to spread them out. I would go for the former except for the fact that I expect most major packages have already finished dealing with this issue.

**Comment by timholy** [comment]
Since this has one approval and we've branched for 1.7, let's do it...

**Pull request merged**